crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -80,25 +85,23 @@ impl RuffSettings {
 
 impl RuffSettingsIndex {
     pub(super) fn new(root: &Path, editor_settings: &ResolvedEditorSettings) -> Self {
-        let mut index = BTreeMap::default();
+        let index = RefCell::new(BTreeMap::new());
```

Comment:
We need some sort of mechanism here because we have to both immutably borrow `index` in `filter_entry` (which _has_ to be a closure) and mutably borrow it in the `for` loop body.

Reply:
Yeah, I don't see any obvious way to avoid using `RefCell` without collecting them all in a vector.